### PR TITLE
Add ML Metadata + Pipelines notebook

### PR DESCRIPTION
**Pull request by sararob**
Adding a new notebook sample that corresponds with this codelab: https://codelabs.developers.google.com/vertex-mlmd-pipelines#0

I've run the linter on this and incorporated changes. Full checklist here:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/master/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/master/docs/CODEOWNERS) file under `# Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb [hunk 1]
```diff
@@ -0,0 +1,1179 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>Break first bullet (after comma) into 2 bullets. Was hard to follow.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>These two arguments conflict. --upgrade would get the latest (1.6) while your pinning it to 1.3.0</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>fyi&gt; It will always be the same bucket if user runs multiple times. Is that what you want?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>Title is plural (Runs), but text is singular (run)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>official brand name is "pandas" -- not capitalized</p><p>same for scikit-learn -- all lowercase</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>I would suggest explaining the steps:</p><p>1. Query BQ for all selecting all rows.</p><p>2. Import the rows into a DataFrame</p><p>3. Randomly sort the DataFrame</p><p>4. Export the DataFrame to CSV file.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>Again, suggest explaining the steps:</p><p>1. Import CSV data into DataFrame</p><p>2. Extract and remove the label column</p><p>3. Split data into train and test data.</p><p>4. Train the model.</p><p>5. Log the metrics.</p><p>6. Save the model artifacts.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>for packages_to_install, you don't use joblib and sklearn in this component.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>uncertain. Wouldn't the user need to copy your datasets into their project? Doesn't look like these are public datasets.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>df -&gt; print(df)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>Pandas -&gt; pandas</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='78'/>

Comment (andrewferlitsch) [inline notebooks/official/ml_metadata/vertex-pipelines-ml-metadata.ipynb:1]: <p>df -&gt; print(df)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='81'/>

**Review by kweinmeister — APPROVED** [review]
